
# Bioinformatics of long read sequencing - ready for the third generation

## Contributors

Amina Echchiki <sup>1,2</sup>, Kamil Jaron <sup>1,2</sup> , Walid Gharib <sup>2,3</sup>

<sup>1: Department of Ecology and Evolution, University of Lausanne, CH-1015 Lausanne</sup>

<sup>2: SIB Swiss Institute of Bioinformatics, CH-1015 Lausanne</sup>

<sup>3: Bioinformatics unit, University of Bern, CH-3012 Bern</sup>


## Practical information 

*When?* October 5th & 6th, 2017 

*Where?* [Irchel Campus](http://www.uzh.ch/en/about/info/sites/irchel.html), University of Zürich, CH-8057 Zürich

*What?* [SIB training event](http://www.sib.swiss/training/upcoming-training-events/2017-longreads02), Hands-on session


## Aim

The aim of this practical session is to get your hands on real data generated from two different long-reads sequencing platforms: Oxford Nanopore (ONT) [MinION](https://nanoporetech.com/products/minion) and Pacific Biosciences (PacBio) [RSII](http://www.pacb.com/products-and-services/pacbio-systems/rsii/). 


## Introduction 

### Biological material

Last year, we analyzed genomic reads from long-reads platforms of the [lambda phage](https://en.wikipedia.org/wiki/Lambda_phage): it is a rather simple organism with small genome size (48kb) that made computations easier but not biologically interesting. 

This year, we propose you to analyze both genomic and transcriptomic data from long-reads sequencing platforms from the fruit fly ([*Drosophila melanogaster*](https://en.wikipedia.org/wiki/Drosophila_melanogaster)), one of the genetically best-known eukaryotic organism. To make computations faster, we focused on assembling the smallest chromosome (chr. 4) and analyzing the transcriptomic reads mapping to it. 

### Long reads technologies 

TODO: compare short and long reads technologies for genome and transcriptome applications 

#### PacBio RSII

TODO: terminology, reads, cool publications, applications, the iso-seq method, perspectives, new applications and innovation 

#### ONT MinION

TODO: terminology, reads, cool publications, applications, perspectives, innovations

## How to connect to the computing platform

TODO: does it change with different OS?

You can use your user credentials that you received for this course (username and password) to activate the Docker container that we built for you (kudos Walid). This image contains all the software you will need for this tutorial. You can then access to the Amazon server (TODO IP adress) we have reserved for your computations. 

TODO add maximal requirements per user.


